websocket.go

```diff
@@ -65,22 +80,22 @@ func (s *Stream) connect(requests ...models.WSRequest) (*websocket.Conn, error)
 		return nil, errors.WithStack(err)
 	}
 
+	err = s.auth(conn)
+
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
 	s.printf("connected to %v", s.url)
 
 	err = s.subscribe(conn, requests)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	lastPong := time.Now()
 	conn.SetPongHandler(func(msg string) error {
-		lastPong = time.Now()
-		if time.Now().Sub(lastPong) > websocketTimeout {
-			// TODO handle this case
-			s.printf("PONG response time has been exceeded")
-		} else {
-			s.printf("PONG")
-		}
+		s.printf("PONG")
+		conn.SetReadDeadline(time.Now().Add(s.wsTimeout))
```

Comment:
what error will come after the timeout?

Reply:
Yes, that needs work. Will check soon-ish.